homeassistant/components/mqtt/lock.py

```diff
@@ -146,10 +152,19 @@ def _prepare_subscribe_topics(self) -> None:
         def message_received(msg: ReceiveMessage) -> None:
             """Handle new MQTT messages."""
             payload = self._value_template(msg.payload)
-            if payload == self._config[CONF_STATE_LOCKED]:
-                self._attr_is_locked = True
-            elif payload == self._config[CONF_STATE_UNLOCKED]:
-                self._attr_is_locked = False
+            valid_states = (
+                self._config[state]
+                for state in (
+                    CONF_STATE_LOCKED,
+                    CONF_STATE_LOCKING,
+                    CONF_STATE_UNLOCKED,
+                    CONF_STATE_UNLOCKING,
+                )
+            )
```

Comment:
May be we should set these in self._valid_states in `self._setup_from_config()` and only of a payload is defined in config.
This way it would be equal to setting

```python
valid states = [CONF_STATE_LOCKED, CONF_STATE_LOCKING, CONF_STATE_UNLOCKED, CONF_STATE_UNLOCKING]
```

Better would be to add to `self._setup_from_config` something like:

```python
self._valid_states = [ config[state] for state in STATE_CONFIG_KEYS ]
```

Don't forget to add type hints ad class level `list[str]`

And add this to the module constants:

```python
STATE_CONFIG_KEYS = [
                    CONF_STATE_LOCKED,
                    CONF_STATE_LOCKING,
                    CONF_STATE_UNLOCKED,
                    CONF_STATE_UNLOCKING,
]
```

Reply:
Done :slightly_smiling_face: